src/reader.rs

```diff
@@ -6511,10 +6511,22 @@ impl<'a> Reader<'a> {
             prefix.push_utfstr(&tok);
             prefix.push_utfstr(&common_prefix);
         } else {
-            // Append just the end of the string.
+            // Collapse parent directories and append end of string
             prefix.push(get_ellipsis_char());
+
             let full = tok + &common_prefix[..];
-            prefix.push_utfstr(&full[full.len() - PREFIX_MAX_LEN..]);
+            let truncated = &full[full.len() - PREFIX_MAX_LEN..];
+            let parts: Vec<&wstr> = truncated.split('/').collect();
+            if parts.len() < 2 {
```

Comment:
I'm wondering if we should also allow `=` and `:` as separators, since we sort of do that when computing completions.
I guess let's wait until we find a use case; the LHS is not usually very large.